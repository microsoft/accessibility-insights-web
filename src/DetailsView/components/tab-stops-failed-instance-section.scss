// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
.tab-stops-failure-instance-section {
    padding-bottom: 58px;

    .title-container {
        .collapsible-control::before {
            position: relative;
            bottom: 2px;
        }
    }

    > h2 {
        margin: 0px;
        font-size: 17px;
        line-height: 24px;
    }
}
