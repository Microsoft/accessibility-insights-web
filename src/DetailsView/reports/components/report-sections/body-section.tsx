// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import * as React from 'react';
import { NamedSFC } from 'common/react/named-sfc';

export const BodySection = NamedSFC('BodySection', ({ children }) => {
    return <body>{children}</body>;
});
