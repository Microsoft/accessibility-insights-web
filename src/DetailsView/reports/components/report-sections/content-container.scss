// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
@import './common-mixins.scss';

.content-container {
    @include contentSize();
    margin-top: 24px;

    h2 {
        margin: 0px;
        font-size: 17px;
        line-height: 24px;
    }

    @media only screen and (max-width: 1000px) {
        margin: 24px 16px 0 16px;
    }
}
