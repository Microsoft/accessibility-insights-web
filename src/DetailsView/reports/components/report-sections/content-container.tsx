// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import * as React from 'react';
import { NamedSFC } from '../../../../common/react/named-sfc';

export const ContentContainer = NamedSFC('ContentSection', ({ children }) => {
    return <div className="content-section">{children}</div>;
});
