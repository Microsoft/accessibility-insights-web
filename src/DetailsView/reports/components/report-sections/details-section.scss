// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

@import '../../../../common/styles/colors.scss';

@mixin boxShadow {
    box-shadow: 0px 0.3px 0.9px rgba(0, 0, 0, 0.108), 0px 1.6px 3.6px rgba(0, 0, 0, 0.132);
    border-radius: 4px;
}

.scan-details-section {
    @include boxShadow();

    background-color: $neutral-0;
    padding: 20px;

    .details-section-list {
        list-style: none;
        font-size: 14px;
        line-height: 16px;
        padding-left: 0px;
        display: flex;
        align-items: flex-start;
        justify-content: center;
        flex-direction: column;
    }

    h3 {
        margin: 0px;
        font-size: 17px;
        line-height: 24px;
    }

    ul :first-child li {
        padding-top: 16px;
    }

    li {
        display: inline-block;
        padding-top: 12px;
        .icon {
            padding-right: 12px;
            position: relative;
            top: 3px;
        }
    }

    .description-text {
        white-space: pre-wrap;
    }
    .screen-reader-only {
        position: absolute;
        left: -10000px;
        top: auto;
        width: 1px;
        height: 1px;
        overflow: hidden;
    }
}
