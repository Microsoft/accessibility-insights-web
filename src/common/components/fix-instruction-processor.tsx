// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import * as React from 'react';
import * as styles from './cards/fix-instruction-color-box.scss';

type ColorMatch = {
    splitIndex: number;
    colorHexValue: string;
};

export class FixInstructionProcessor {
    private readonly colorValueMatcher = `(#[0-9a-f]{6})`;
    private readonly foregroundColorText = 'foreground color: ';
    private readonly foregroundRegExp = new RegExp(
        `${this.foregroundColorText}${this.colorValueMatcher}`,
        'i',
    );
    private readonly backgroundColorText = 'background color: ';
    private readonly backgroundRegExp = new RegExp(
        `${this.backgroundColorText}${this.colorValueMatcher}`,
        'i',
    );

    public process(fixInstruction: string): JSX.Element {
        const foregroundMatch = this.getColorMatch(fixInstruction, this.foregroundRegExp);
        const backgroundMatch = this.getColorMatch(fixInstruction, this.backgroundRegExp);

        const matches = [foregroundMatch, backgroundMatch];

        return this.splitFixInstruction(fixInstruction, matches);
    }

    private getColorMatch(fixInstruction: string, colorRegex: RegExp): ColorMatch {
        if (!colorRegex.test(fixInstruction)) {
            return null;
        }

        const match = colorRegex.exec(fixInstruction);
        const colorHexValue = match[1];

        return {
            splitIndex: match.index + this.foregroundColorText.length + colorHexValue.length,
            colorHexValue,
        };
    }

    private splitFixInstruction(fixInstruction: string, matches: ColorMatch[]): JSX.Element {
        const properMatches = matches
            .filter(current => current != null)
            .sort((a, b) => a.splitIndex - b.splitIndex);

        if (properMatches.length === 0) {
            return <>{fixInstruction}</>;
        }

        let insertionIndex = 0;
        let keyIndex = 0;

        const result: JSX.Element[] = [];

        properMatches.forEach(match => {
            const endIndex = match.splitIndex - match.colorHexValue.length;
            const substring = fixInstruction.substring(insertionIndex, endIndex);

            result.push(<span key={`instruction-split-${keyIndex++}`}>{substring}</span>);

            result.push(this.createColorBox(match.colorHexValue, keyIndex++));

            insertionIndex = endIndex;
        });

        const coda = fixInstruction.substr(insertionIndex);

        result.push(<span key={`instruction-split-${keyIndex++}`}>{coda}</span>);

        return (
            <>
                <span aria-hidden="true">{result}</span>
                <span className={styles.screenReaderOnly}>{fixInstruction}</span>
            </>
        );
    }

    private createColorBox(colorHexValue: string, keyIndex: number): JSX.Element {
        return (
            <span
                key={`instruction-split-${keyIndex}`}
                className={styles.fixInstructionColorBox}
                style={{ backgroundColor: colorHexValue }}
            />
        );
    }
}
