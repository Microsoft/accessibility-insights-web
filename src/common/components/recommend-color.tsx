// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

const MAXCOLORCOMPONENT: number = 255;
const MINCOLORCOMPONENT: number = 0;

export class RecommendColor {
    private _sentence: string;

    constructor(fore: string, back: string, contrast: number) {
        this._sentence = this.recommend(this.hexToRGB(fore), this.hexToRGB(back), contrast);
    }

    public get sentence(): string {
        return this._sentence;
    }

    private rgbToHex(r: number, g: number, b: number): string {
        return '#' + ((1 << 24) + (r << 16) + (g << 8) + b).toString(16).slice(1);
    }

    private hexToRGB(hex: string): number[] {
        const result = /^#?([a-f\d]{2})([a-f\d]{2})([a-f\d]{2})$/i.exec(hex);
        return [parseInt(result[1], 16), parseInt(result[2], 16), parseInt(result[3], 16)];
    }

    private getLuminance(color: number[]): number {
        const a = [color[0], color[1], color[2]].map(function (v) {
            v /= 255;
            return v <= 0.03928 ? v / 12.92 : Math.pow((v + 0.055) / 1.055, 2.4);
        });
        return a[0] * 0.2126 + a[1] * 0.7152 + a[2] * 0.0722;
    }

    private darkenLighten(color: number[], shade: boolean): number[] {
        let num: number = 0.95;
        if (shade) {
            num = 1.05;
            if (color[0] === 0 && color[1] === 0 && color[2] === 0) {
                color = [10, 10, 10];
            }
        }

        for (let i = 0; i < 3; i++) {
            let component: number = color[i];
            if (component * num <= MAXCOLORCOMPONENT && component * num >= MINCOLORCOMPONENT) {
                color[i] = Math.floor(component * num);
            } else if (shade) {
                color[i] = MAXCOLORCOMPONENT;
            } else {
                color[i] = MINCOLORCOMPONENT;
            }
        }

        return color;
    }

    private recommendAColor(
        contrast: number,
        toBeChanged: number[],
        originalPair: number,
        toLightenDarken: boolean,
        colorContrast: number,
    ): object {
        let preContrast: number = -1;

        while (contrast < colorContrast && contrast !== preContrast) {
            preContrast = contrast;
            toBeChanged = this.darkenLighten(toBeChanged, toLightenDarken);
            if (toLightenDarken) {
                contrast = (this.getLuminance(toBeChanged) + 0.05) / (originalPair + 0.05);
            } else {
                contrast = (originalPair + 0.05) / (this.getLuminance(toBeChanged) + 0.05);
            }
        }

        return [toBeChanged, contrast];
    }

    private recommend(fore: number[], back: number[], colorContrast: number) {
        let lighter: number, darker: number, contrast: number;
        let foreLum = this.getLuminance(fore);
        let backLum = this.getLuminance(back);
        lighter = Math.max(foreLum, backLum);
        darker = lighter === foreLum ? backLum : foreLum;
        contrast = (lighter + 0.05) / (darker + 0.05);

        let lightened: number[], darkened: number[];

        if (lighter === foreLum) {
            lightened = fore;
            darkened = back;
        } else {
            lightened = back;
            darkened = fore;
        }

        let recommendLightened: object, recommendDarkened: object;
        recommendLightened = this.recommendAColor(contrast, lightened, darker, true, colorContrast);
        recommendDarkened = this.recommendAColor(contrast, darkened, lighter, false, colorContrast);

        let sentence: string = '',
            foreRec: string,
            backRec: string;

        if (lighter === foreLum) {
            if (recommendLightened[1] > colorContrast) {
                foreRec = this.rgbToHex(
                    recommendLightened[0][0],
                    recommendLightened[0][1],
                    recommendLightened[0][2],
                );
                sentence =
                    ' Use foreground color: ' +
                    foreRec +
                    ' to meet a contrast ratio of ' +
                    recommendLightened[1].toFixed(2) +
                    ':1.';
            }
            if (recommendDarkened[1] > colorContrast) {
                backRec = this.rgbToHex(
                    recommendDarkened[0][0],
                    recommendDarkened[0][1],
                    recommendDarkened[0][2],
                );
                sentence =
                    sentence +
                    ' Use background color: ' +
                    backRec +
                    ' to meet a contrast ratio of ' +
                    recommendDarkened[1].toFixed(2) +
                    ':1.';
            }
        } else {
            if (recommendDarkened[1] > colorContrast) {
                foreRec = this.rgbToHex(
                    recommendDarkened[0][0],
                    recommendDarkened[0][1],
                    recommendDarkened[0][2],
                );
                sentence =
                    ' Use foreground color: ' +
                    foreRec +
                    ' to meet a contrast ratio of ' +
                    recommendDarkened[1].toFixed(2) +
                    ':1.';
            }
            if (recommendLightened[1] > colorContrast) {
                backRec = this.rgbToHex(
                    recommendLightened[0][0],
                    recommendLightened[0][1],
                    recommendLightened[0][2],
                );
                sentence =
                    sentence +
                    ' Use background color: ' +
                    backRec +
                    ' to meet a contrast ratio of ' +
                    recommendLightened[1].toFixed(2) +
                    ':1.';
            }
        }

        return sentence;
    }
}
