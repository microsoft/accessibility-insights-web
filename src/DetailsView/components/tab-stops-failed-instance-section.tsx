// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

import { ResultSectionTitle } from 'common/components/cards/result-section-title';
import { HeadingElementForLevel, HeadingLevel } from 'common/components/heading-element-for-level';
import { FeatureFlags } from 'common/feature-flags';
import { NamedFC } from 'common/react/named-fc';
import { FeatureFlagStoreData } from 'common/types/store-data/feature-flag-store-data';
import { TabStopRequirementState } from 'common/types/store-data/visualization-scan-result-data';
import * as styles from 'DetailsView/components/tab-stops-failed-instance-section.scss';
import { requirements } from 'DetailsView/components/tab-stops/requirements';
import { TabStopsInstanceSectionPropsFactory } from 'DetailsView/components/tab-stops/tab-stops-instance-section-props-factory';
import { TabStopsFailedCounter } from 'DetailsView/tab-stops-failed-counter';
import {
    TabStopsRequirementsWithInstances,
    TabStopsRequirementsWithInstancesDeps,
} from 'DetailsView/tab-stops-requirements-with-instances';
import * as React from 'react';

export type TabStopsFailedInstanceSectionDeps = TabStopsRequirementsWithInstancesDeps & {
    tabStopsFailedCounter: TabStopsFailedCounter;
    tabStopsInstanceSectionPropsFactory: TabStopsInstanceSectionPropsFactory;
    getNextHeadingLevel: (headingLevel: HeadingLevel) => HeadingLevel;
};

export interface TabStopsFailedInstanceSectionProps {
    deps: TabStopsFailedInstanceSectionDeps;
    tabStopRequirementState: TabStopRequirementState;
    alwaysRenderSection: boolean;
    sectionHeadingLevel: HeadingLevel;
    featureFlagStoreData: FeatureFlagStoreData;
}

export const tabStopsFailedInstanceSectionAutomationId = 'tab-stops-failure-instance-section';

export const TabStopsFailedInstanceSection = NamedFC<TabStopsFailedInstanceSectionProps>(
    'TabStopsFailedInstanceSection',
    props => {
        const results = [];

        for (const [requirementId, data] of Object.entries(props.tabStopRequirementState)) {
            if (data.status !== 'fail') {
                continue;
            }
            const displayAutomatedInfo =
                props.featureFlagStoreData != null &&
                props.featureFlagStoreData[FeatureFlags.tabStopsAutomation];

            results.push({
                id: requirementId,
                name: requirements(displayAutomatedInfo)[requirementId].name,
                description: requirements(displayAutomatedInfo)[requirementId].description,
                instances: data.instances,
                isExpanded: data.isExpanded,
            });
        }

        const totalFailedInstancesCount: number =
            props.deps.tabStopsFailedCounter.getTotalFailed(results);

        if (!props.alwaysRenderSection && totalFailedInstancesCount === 0) {
            return null;
        }

        const instanceSectionProps = props.deps.tabStopsInstanceSectionPropsFactory({
            headingLevel: props.deps.getNextHeadingLevel(props.sectionHeadingLevel),
            results,
            tabStopRequirementState: props.tabStopRequirementState,
            deps: props.deps,
        });

        return (
            <div
                className={styles.tabStopsFailureInstanceSection}
                data-automation-id={tabStopsFailedInstanceSectionAutomationId}
            >
                <HeadingElementForLevel headingLevel={props.sectionHeadingLevel}>
                    <ResultSectionTitle
                        title="Failed instances"
                        badgeCount={totalFailedInstancesCount}
                        outcomeType="fail"
                        titleSize="title"
                    />
                </HeadingElementForLevel>
                <TabStopsRequirementsWithInstances {...instanceSectionProps} />
            </div>
        );
    },
);
