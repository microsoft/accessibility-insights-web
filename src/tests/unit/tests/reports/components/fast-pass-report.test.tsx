// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { render } from '@testing-library/react';
import { FixInstructionProcessor } from 'common/components/fix-instruction-processor';
import { RecommendColor } from 'common/components/recommend-color';
import * as React from 'react';
import {
    FastPassReport,
    FastPassReportDeps,
    FastPassReportProps,
} from 'reports/components/fast-pass-report';
import { Mock } from 'typemoq';
import { TabStopsFailedInstanceSection } from '../../../../../DetailsView/components/tab-stops-failed-instance-section';
import { FastPassReportSummary } from '../../../../../reports/components/fast-pass-report-summary';
import { AutomatedChecksHeaderSection } from '../../../../../reports/components/report-sections/automated-checks-header-section';
import { BodySection } from '../../../../../reports/components/report-sections/body-section';
import { ContentContainer } from '../../../../../reports/components/report-sections/content-container';
import { DetailsSection } from '../../../../../reports/components/report-sections/details-section';
import { FastPassReportAutomatedChecksResults } from '../../../../../reports/components/report-sections/fast-pass-report-automated-checks-results';
import { FastPassResultsTitleSection } from '../../../../../reports/components/report-sections/fast-pass-results-title-section';
import { FastPassTitleSection } from '../../../../../reports/components/report-sections/fast-pass-title-section';
import { FooterText } from '../../../../../reports/components/report-sections/footer-text';
import { ReportFooter } from '../../../../../reports/components/report-sections/report-footer';
import { ResultsContainer } from '../../../../../reports/components/report-sections/results-container';
import { TabStopsChecksSectionWrapper } from '../../../../../reports/components/report-sections/tab-stops-checks-section-wrapper';
import { WebReportHead } from '../../../../../reports/components/web-report-head';
import { mockReactComponents } from '../../../mock-helpers/mock-module-helpers';
import { exampleUnifiedStatusResults } from '../../common/components/cards/sample-view-model-data';
jest.mock('../../../../../reports/components/web-report-head');
jest.mock('../../../../../DetailsView/components/tab-stops-failed-instance-section');
jest.mock('../../../../../reports/components/report-sections/body-section');
jest.mock('../../../../../reports/components/fast-pass-report-summary');
jest.mock('../../../../../reports/components/report-sections/automated-checks-header-section');
jest.mock('../../../../../reports/components/report-sections/content-container');
jest.mock('../../../../../reports/components/report-sections/fast-pass-title-section');
jest.mock('../../../../../reports/components/report-sections/details-section');
jest.mock(
    '../../../../../reports/components/report-sections/fast-pass-report-automated-checks-results',
);
jest.mock('../../../../../reports/components/report-sections/report-footer');
jest.mock('../../../../../reports/components/report-sections/footer-text');
jest.mock('../../../../../reports/components/report-sections/results-container');
jest.mock('../../../../../reports/components/report-sections/fast-pass-results-title-section');
jest.mock('../../../../../reports/components/report-sections/tab-stops-checks-section-wrapper');

describe(FastPassReport.displayName, () => {
    mockReactComponents([
        WebReportHead,
        BodySection,
        AutomatedChecksHeaderSection,
        ContentContainer,
        FastPassTitleSection,
        DetailsSection,
        FastPassReportSummary,
        ResultsContainer,
        FastPassResultsTitleSection,
        FastPassReportAutomatedChecksResults,
        TabStopsFailedInstanceSection,
        TabStopsChecksSectionWrapper,
        ReportFooter,
        FooterText,
    ]);
    it('renders', () => {
        const pageTitle = 'page-title';
        const pageUrl = 'url:target-page';
        const scanDate = new Date(Date.UTC(0, 1, 2, 3));
        const getScriptStub = () => '';
        const getGuidanceTagsStub = () => [];
        const fixInstructionProcessorMock = Mock.ofType(FixInstructionProcessor);
        const recommendColorMock = Mock.ofType(RecommendColor);
        const toolData = {
            scanEngineProperties: {
                name: 'engine-name',
                version: 'engine-version',
            },
            applicationProperties: {
                name: 'app-name',
                version: 'app-version',
                environmentName: 'environmentName',
            },
        };
        const targetAppInfo = { name: 'app' };

        const props: FastPassReportProps = {
            deps: {} as FastPassReportDeps,
            fixInstructionProcessor: fixInstructionProcessorMock.object,
            recommendColor: recommendColorMock.object,
            pageTitle,
            pageUrl,
            description: 'test description',
            toolData,
            scanResult: {
                passes: [],
                violations: [],
                inapplicable: [],
                incomplete: [],
                timestamp: 'today',
                targetPageTitle: pageTitle,
                targetPageUrl: pageUrl,
            },
            toUtcString: () => '',
            getCollapsibleScript: getScriptStub,
            getGuidanceTagsFromGuidanceLinks: getGuidanceTagsStub,
            results: {
                automatedChecks: {
                    cards: exampleUnifiedStatusResults,
                    visualHelperEnabled: true,
                    allCardsCollapsed: true,
                },
                tabStops: {
                    'keyboard-traps': {
                        status: 'pass',
                        instances: [{ id: 'test-id-2', description: 'test desc 2' }],
                        isExpanded: false,
                    },
                    'tab-order': {
                        status: 'fail',
                        instances: [{ id: 'test-id-4', description: 'test desc 4' }],
                        isExpanded: false,
                    },
                },
            },
            userConfigurationStoreData: null,
            targetAppInfo,
            shouldAlertFailuresCount: false,
            scanMetadata: {
                toolData,
                targetAppInfo,
                timespan: {
                    scanComplete: scanDate,
                },
            },
            sectionHeadingLevel: 3,
        };

        const renderResult = render(<FastPassReport {...props} />);

        expect(renderResult.asFragment()).toMatchSnapshot();
    });
});
