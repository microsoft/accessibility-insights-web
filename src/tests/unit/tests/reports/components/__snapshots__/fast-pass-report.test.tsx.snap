// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` renders 1`] = `
"<Fragment>
  <WebReportHead />
  <BodySection>
    <AutomatedChecksHeaderSection deps={{...}} fixInstructionProcessor={{...}} recommendColor={{...}} pageTitle=\\"page-title\\" pageUrl=\\"url:target-page\\" description=\\"test description\\" toolData={{...}} scanResult={{...}} toUtcString={[Function: toUtcString]} getCollapsibleScript={[Function: getScriptStub]} getGuidanceTagsFromGuidanceLinks={[Function: getGuidanceTagsStub]} results={{...}} userConfigurationStoreData={{...}} targetAppInfo={{...}} shouldAlertFailuresCount={false} scanMetadata={{...}} sectionHeadingLevel={3} />
    <ContentSection>
      <FastPassTitleSection />
      <DetailsSection deps={{...}} fixInstructionProcessor={{...}} recommendColor={{...}} pageTitle=\\"page-title\\" pageUrl=\\"url:target-page\\" description=\\"test description\\" toolData={{...}} scanResult={{...}} toUtcString={[Function: toUtcString]} getCollapsibleScript={[Function: getScriptStub]} getGuidanceTagsFromGuidanceLinks={[Function: getGuidanceTagsStub]} results={{...}} userConfigurationStoreData={{...}} targetAppInfo={{...}} shouldAlertFailuresCount={false} scanMetadata={{...}} sectionHeadingLevel={3} />
      <FastPassReportSummary deps={{...}} fixInstructionProcessor={{...}} recommendColor={{...}} pageTitle=\\"page-title\\" pageUrl=\\"url:target-page\\" description=\\"test description\\" toolData={{...}} scanResult={{...}} toUtcString={[Function: toUtcString]} getCollapsibleScript={[Function: getScriptStub]} getGuidanceTagsFromGuidanceLinks={[Function: getGuidanceTagsStub]} results={{...}} userConfigurationStoreData={{...}} targetAppInfo={{...}} shouldAlertFailuresCount={false} scanMetadata={{...}} sectionHeadingLevel={3} featureFlagStoreData={[undefined]} />
      <ResultsContainer deps={{...}} fixInstructionProcessor={{...}} recommendColor={{...}} pageTitle=\\"page-title\\" pageUrl=\\"url:target-page\\" description=\\"test description\\" toolData={{...}} scanResult={{...}} toUtcString={[Function: toUtcString]} getCollapsibleScript={[Function: getScriptStub]} getGuidanceTagsFromGuidanceLinks={[Function: getGuidanceTagsStub]} results={{...}} userConfigurationStoreData={{...}} targetAppInfo={{...}} shouldAlertFailuresCount={false} scanMetadata={{...}} sectionHeadingLevel={3}>
        <FastPassResultsTitleSection title=\\"Automated checks\\" />
        <FastPassReportAutomatedChecksResults deps={{...}} fixInstructionProcessor={{...}} recommendColor={{...}} pageTitle=\\"page-title\\" pageUrl=\\"url:target-page\\" description=\\"test description\\" toolData={{...}} scanResult={{...}} toUtcString={[Function: toUtcString]} getCollapsibleScript={[Function: getScriptStub]} getGuidanceTagsFromGuidanceLinks={[Function: getGuidanceTagsStub]} results={{...}} userConfigurationStoreData={{...}} targetAppInfo={{...}} shouldAlertFailuresCount={false} scanMetadata={{...}} sectionHeadingLevel={3} />
        <FastPassResultsTitleSection title=\\"Tab stops\\" />
        <TabStopsFailedInstanceSection sectionHeadingLevel={3} deps={{...}} featureFlagStoreData={[undefined]} tabStopRequirementState={{...}} alwaysRenderSection={true} />
        <TabStopsChecksSectionWrapper checksSection={[Function (anonymous)] { displayName: 'IncompleteChecksSection' }} tabStops={{...}} featureFlagStoreData={[undefined]} deps={{...}} fixInstructionProcessor={{...}} recommendColor={{...}} pageTitle=\\"page-title\\" pageUrl=\\"url:target-page\\" description=\\"test description\\" toolData={{...}} scanResult={{...}} toUtcString={[Function: toUtcString]} getCollapsibleScript={[Function: getScriptStub]} getGuidanceTagsFromGuidanceLinks={[Function: getGuidanceTagsStub]} results={{...}} userConfigurationStoreData={{...}} targetAppInfo={{...}} shouldAlertFailuresCount={false} scanMetadata={{...}} sectionHeadingLevel={3} />
        <TabStopsChecksSectionWrapper checksSection={[Function (anonymous)] { displayName: 'PassedChecksSection' }} tabStops={{...}} featureFlagStoreData={[undefined]} deps={{...}} fixInstructionProcessor={{...}} recommendColor={{...}} pageTitle=\\"page-title\\" pageUrl=\\"url:target-page\\" description=\\"test description\\" toolData={{...}} scanResult={{...}} toUtcString={[Function: toUtcString]} getCollapsibleScript={[Function: getScriptStub]} getGuidanceTagsFromGuidanceLinks={[Function: getGuidanceTagsStub]} results={{...}} userConfigurationStoreData={{...}} targetAppInfo={{...}} shouldAlertFailuresCount={false} scanMetadata={{...}} sectionHeadingLevel={3} />
      </ResultsContainer>
    </ContentSection>
    <ReportFooter>
      <FooterText deps={{...}} fixInstructionProcessor={{...}} recommendColor={{...}} pageTitle=\\"page-title\\" pageUrl=\\"url:target-page\\" description=\\"test description\\" toolData={{...}} scanResult={{...}} toUtcString={[Function: toUtcString]} getCollapsibleScript={[Function: getScriptStub]} getGuidanceTagsFromGuidanceLinks={[Function: getGuidanceTagsStub]} results={{...}} userConfigurationStoreData={{...}} targetAppInfo={{...}} shouldAlertFailuresCount={false} scanMetadata={{...}} sectionHeadingLevel={3} />
    </ReportFooter>
  </BodySection>
</Fragment>"
`;
